backsplash-core/src/main/scala/com/rasterfoundry/backsplash/Implicits.scala

```diff
@@ -23,36 +25,95 @@ trait Implicits
     extends ToTmsReificationOps
     with ToExtentReificationOps
     with ToHasRasterExtentsOps
-    with ToProjectStoreOps {
+    with ToProjectStoreOps
+    with ToToolStoreOps {
+
+  implicit def paintableToolTmsReification[B: TmsReification] =
+    new TmsReification[PaintableTool[B]] {
+      def kind(self: PaintableTool[B]): MamlKind = MamlKind.Image
+
+      def tmsReification(self: PaintableTool[B], buffer: Int)(
+          implicit contextShfit: ContextShift[IO]
+      ): (Int, Int, Int) => IO[Literal] = (z: Int, x: Int, y: Int) => {
+        val extent = BacksplashImage.tmsLevels(z).mapTransform.keyToExtent(x, y)
+        val layerEval = LayerTms.apply(IO.pure(self.expr),
+                                       IO.pure(self.paramMap),
+                                       self.interpreter)
+        layerEval(z, x, y) map {
+          case Valid(tile) => RasterLit(Raster(self.painter(tile), extent))
+          case Invalid(e)  => throw new Exception(s"Unresolvable tile: $e")
+        }
+      }
+    }
+
+  implicit def paintableToolExtentReification[B: ExtentReification] =
+    new ExtentReification[PaintableTool[B]] {
+      def kind(self: PaintableTool[B]): MamlKind = MamlKind.Image
+      def extentReification(self: PaintableTool[B])(
+          implicit contextShift: ContextShift[IO]) = {
+        val layerEval = LayerExtent.apply(IO.pure(self.expr),
+                                          IO.pure(self.paramMap),
+                                          self.interpreter)
+        (extent: Extent, cellSize: CellSize) =>
+          layerEval(extent, cellSize) map {
+            case Valid(tile) =>
+              if (self.paint) {
+                RasterLit(Raster(self.painter(tile), extent))
+              } else {
+                RasterLit(Raster(tile, extent))
+              }
+            case Invalid(e) => throw new Exception(s"Unresolvable extent: $e")
+          }
+      }
+    }
+
+  implicit def paintableToolHasRasterExtents[B: HasRasterExtents] =
```

Comment:
I think that this implicit is likely in violation of the semantics expected by GTServer code. The `PaintableTool` is not, itself, a valid parameter in this context but rather a container for holding pieces which, themselves, could serve as valid parameters.

For instance, in histogram generation, what we want is for an intersection between all parameters so that a meaningful computation area can be inferred and checked once the raster is computed. This will avoid that step by concatenating together the various `RasterExtent`s. It might make more sense to just have this type bundle GT Server calls in a handy way... I'm going to throw together a small commit with that in mind

Reply:
"container for holding pieces" is also what a `BacksplashMosaic` is -- I'm not really clear on the distinction you're drawing here.